ValueCodable.playground/Contents.swift

```diff
@@ -108,18 +108,58 @@ enum Zodiac: UnicodeScalar, ValueCodable {
     case Pisces = "♓"
 }
 
+protocol DummyProtocolObject: ValueCodable {
+    var x: Int { get }
+}
+
+extension DummyProtocolObject {
+    func encodeDynamic(with coder: NSCoder, forKey key: String) {
+        coder.encodeDynamicValue(self, asType: self.dynamicType, forKey: key)
```

Comment:
This line is the most suspect of them all. We're calling a generic function with a type bound to something that isn't known until runtime. I'm sorta surprised this works at all. This is taking advantage of the whole "Swift doesn't always specialize generic functions" thing that's usually annoying.